server/etcdserver/api/v3rpc/maintenance.go

```diff
@@ -125,20 +125,18 @@ func (ms *maintenanceServer) Snapshot(sr *pb.SnapshotRequest, srv pb.Maintenance
 	total := snap.Size()
 	size := humanize.Bytes(uint64(total))
 
+	// buffer just holds read bytes from stream
+	// response size is multiple of OS page size, fetched in boltdb
+	// e.g. 4*1024
+	buf := make([]byte, snapshotSendBufferSize)
```

Comment:
Reusing buffer is nice, but please reset it after each use.

Reply:
updated.

Use the following pattern:

```
for idx := range buf {
    buf[idx] = 0
}
```

It will be 

```bash
$ make
$ go tool objdump -s "go.etcd.io/etcd/server/v3/etcdserver/api/v3rpc" bin/etcd | grep maintenance.go:165
      maintenance.go:165    0xd5ce72                488b842468010000                MOVQ 0x168(SP), AX
      maintenance.go:165    0xd5ce7a                bb00800000                      MOVL $0x8000, BX
      maintenance.go:165    0xd5ce7f                90                              NOPL
      maintenance.go:165    0xd5ce80                e81b1e71ff                      CALL runtime.memclrNoHeapPointers(SB)
```

 REF: https://github.com/golang/go/issues/5373